packages/wonder-blocks-icon-button/components/icon-button.test.js

```diff
@@ -30,4 +37,61 @@ describe("IconButton", () => {
             ),
         ).toThrowError("Light is only supported for primary IconButtons");
     });
+
+    test("client-side navigation", () => {
+        const wrapper = mount(
+            <MemoryRouter>
+                <div>
+                    <IconButton
+                        icon={icons.search}
+                        aria-label="search"
+                        testId="icon-button"
+                        href="/foo"
+                        clientNav={true}
+                    />
+                    <Switch>
+                        <Route path="/foo">
+                            <div id="foo">Hello, world!</div>
+                        </Route>
+                    </Switch>
+                </div>
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper
+            .find(`[data-test-id="icon-button"]`)
+            .first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(true);
+    });
+
+    test("client-side navigation without 'clientNav' prop fails", () => {
+        const wrapper = mount(
+            <MemoryRouter>
+                <div>
+                    <IconButton
+                        icon={icons.search}
+                        aria-label="search"
+                        testId="icon-button"
+                        href="/foo"
+                    />
+                    <Switch>
+                        <Route path="/foo">
+                            <div id="foo">Hello, world!</div>
+                        </Route>
+                    </Switch>
+                </div>
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper
+            .find(`[data-test-id="icon-button"]`)
+            .first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(false);
```

Comment:
I'd love to see `// Arrange`, `// Act`, `// Assert` structure in these tests to make it clear what the test is doing.